processing/src/main/java/io/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -55,7 +56,7 @@
    */
   private static final int NUM_BYTES_IN_TEXT_FLOAT = 8;
 
-  public Map<String, ColumnAnalysis> analyze(QueryableIndex index)
+  public Map<String, ColumnAnalysis> analyze(QueryableIndex index, SegmentMetadataQuery query)
```

Comment:
@gianm @xvrl changed SegmentAnalyzer functions to accept the flag set instead of query